Intermitent StackVisualiserViewTest test failure

### Summary

StackVisualiserViewTest fails sometimes, dependent on test ordering.

### Steps To Reproduce

In my centos7 VM

`pytest -n 1 --count 1 --randomly-seed=274074171 -k StackVisualiserViewTest -v`

Seems to reliably trigger the failure. 

On my Ubuntu 20.04 laptop I need

`pytest -n 1 --count 1 --randomly-seed=3153455096 -k StackVisualiserViewTest -v`

Otherwise you can find a bad seed with
`while pytest -n 1 --count 1  -k StackVisualiserViewTest -v ; do echo ; done`

### Expected Behaviour

Tests pass

### Current Behaviour

`[gw0] [100%] PASSED mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_closeEvent_deletes_images_with_proj180_user_declined Fatal Python error: Segmentation fault`


### Context

Happens on master since the python 3.8 merge

### Failure Logs

```
(midev) [sam@localhost mantidimaging]$ pytest -n 1 --count 1 --randomly-seed=274074171 -k StackVisualiserViewTest -v
===================================================================== test session starts ======================================================================
platform linux -- Python 3.8.6, pytest-6.2.1, py-1.10.0, pluggy-0.13.1 -- /home/sam/miniconda3/envs/midev/bin/python
cachedir: .pytest_cache
Using --randomly-seed=274074171
rootdir: /home/sam/git/mantidimaging
plugins: forked-1.3.0, xdist-2.2.0, repeat-0.9.1, randomly-3.5.0, cov-2.10.1
[gw0] linux Python 3.8.6 cwd: /home/sam/git/mantidimaging
[gw0] Python 3.8.6 | packaged by conda-forge | (default, Dec 26 2020, 05:05:16)  -- [GCC 9.3.0]
gw0 [5]
scheduling tests via LoadScheduling

mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_closeEvent_deletes_images 
[gw0] [ 20%] PASSED mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_closeEvent_deletes_images 
mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_roi_changed_callback 
[gw0] [ 40%] PASSED mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_roi_changed_callback 
mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_name 
[gw0] [ 60%] PASSED mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_name 
mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_closeEvent_deletes_images_with_proj180_user_accepts 
[gw0] [ 80%] PASSED mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_closeEvent_deletes_images_with_proj180_user_accepts 
mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_closeEvent_deletes_images_with_proj180_user_declined 
[gw0] [100%] PASSED mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_closeEvent_deletes_images_with_proj180_user_declined Fatal Python error: Segmentation fault

Thread 0x00007f97ce0c1700 (most recent call first):
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/execnet/gateway_base.py", line 400 in read
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/execnet/gateway_base.py", line 432 in from_io
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/execnet/gateway_base.py", line 967 in _thread_receiver
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/execnet/gateway_base.py", line 220 in run
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/execnet/gateway_base.py", line 285 in _perform_spawn

Current thread 0x00007f97d5833740 (most recent call first):
  File "/home/sam/git/mantidimaging/mantidimaging/test_helpers/start_qapplication.py", line 58 in tearDownClass
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/unittest.py", line 175 in fixture
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/fixtures.py", line 941 in _teardown_yield_fixture
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/fixtures.py", line 1027 in finish
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 395 in _callfinalizers
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 405 in _teardown_with_finalization
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 387 in _pop_and_teardown
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 427 in _teardown_towards
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 419 in teardown_exact
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 175 in pytest_runtest_teardown
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 255 in <lambda>
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 311 in from_call
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 254 in call_runtest_hook
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 215 in call_and_report
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 127 in runtestprotocol
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/runner.py", line 109 in pytest_runtest_protocol
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/xdist/remote.py", line 89 in run_one_test
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/xdist/remote.py", line 75 in pytest_runtestloop
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/main.py", line 323 in _main
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/main.py", line 269 in wrap_session
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/_pytest/main.py", line 316 in pytest_cmdline_main
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/xdist/remote.py", line 239 in <module>
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/execnet/gateway_base.py", line 1084 in executetask
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/execnet/gateway_base.py", line 220 in run
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/execnet/gateway_base.py", line 285 in _perform_spawn
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/execnet/gateway_base.py", line 267 in integrate_as_primary_thread
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/execnet/gateway_base.py", line 1060 in serve
  File "/home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/execnet/gateway_base.py", line 1554 in serve
  File "<string>", line 8 in <module>
  File "<string>", line 1 in <module>

[gw0] node down: Not properly terminated
[gw0] [100%] FAILED mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_closeEvent_deletes_images_with_proj180_user_declined 

replacing crashed worker gw0
[gw1] linux Python 3.8.6 cwd: /home/sam/git/mantidimaging
[gw1] Python 3.8.6 | packaged by conda-forge | (default, Dec 26 2020, 05:05:16)  -- [GCC 9.3.0]
gw1 ok
=========================================================================== FAILURES ===========================================================================
_________________________________________________ mantidimaging/gui/windows/stack_visualiser/test/view_test.py _________________________________________________
[gw0] linux -- Python 3.8.6 /home/sam/miniconda3/envs/midev/bin/python
worker 'gw0' crashed while running 'mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_closeEvent_deletes_images_with_proj180_user_declined'
======================================================================= warnings summary =======================================================================
../../miniconda3/envs/midev/lib/python3.8/site-packages/skimage/util/dtype.py:226
../../miniconda3/envs/midev/lib/python3.8/site-packages/skimage/util/dtype.py:226
  /home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/skimage/util/dtype.py:226: DeprecationWarning: Converting `np.inexact` or `np.floating` to a dtype is deprecated. The current result is `float64` which is not strictly correct.
    dtypeobj_out = np.dtype(dtype)

../../miniconda3/envs/midev/lib/python3.8/site-packages/tomopy/recon/acceleration.py:57
../../miniconda3/envs/midev/lib/python3.8/site-packages/tomopy/recon/acceleration.py:57
  /home/sam/miniconda3/envs/midev/lib/python3.8/site-packages/tomopy/recon/acceleration.py:57: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=================================================================== short test summary info ====================================================================
FAILED mantidimaging/gui/windows/stack_visualiser/test/view_test.py::StackVisualiserViewTest::test_closeEvent_deletes_images_with_proj180_user_declined
=========================================================== 1 failed, 5 passed, 4 warnings in 5.13s ============================================================

```

### Screenshot(s)

<!--- [Optional] Include screenshots that illustrate the problem. -->

--- a/.github/workflows/conda.yml
+++ b/.github/workflows/conda.yml
@@ -43,15 +43,15 @@ jobs:
         run: |
           mypy --ignore-missing-imports mantidimaging
 
-      - name: docs
-        shell: bash -l {0}
-        run: |
-          python setup.py docs
-
       - name: pytest
         shell: bash -l {0}
         run: |
           xvfb-run --auto-servernum python -m pytest --cov --cov-report=xml -n auto --count 10
+
+      - name: docs
+        shell: bash -l {0}
+        run: |
+          python setup.py docs
 
       - name: Coveralls
         shell: bash -l {0}

--- a/.github/workflows/cos7_testing.yml
+++ b/.github/workflows/cos7_testing.yml
@@ -43,14 +43,14 @@ jobs:
         command: mypy --ignore-missing-imports mantidimaging
         label: centos7
 
-    - name: docs
-      uses: ./.github/actions/test
-      with:
-        command: python setup.py docs
-        label: centos7
-
     - name: pytest
       uses: ./.github/actions/test
       with:
         command: xvfb-run pytest -n auto --count 10
+        label: centos7
+
+    - name: docs
+      uses: ./.github/actions/test
+      with:
+        command: python setup.py docs
         label: centos7

--- a/.github/workflows/u18_testing.yml
+++ b/.github/workflows/u18_testing.yml
@@ -39,12 +39,12 @@ jobs:
       with:
         command: mypy --ignore-missing-imports mantidimaging
 
-    - name: docs
-      uses: ./.github/actions/test
-      with:
-        command: python setup.py docs
-
     - name: pytest
       uses: ./.github/actions/test
       with:
         command: xvfb-run pytest -n auto --count 10
+
+    - name: docs
+      uses: ./.github/actions/test
+      with:
+        command: python setup.py docs

--- a/mantidimaging/gui/windows/stack_visualiser/test/view_test.py
+++ b/mantidimaging/gui/windows/stack_visualiser/test/view_test.py
@@ -17,6 +17,7 @@ from mantidimaging.gui.windows.stack_visualiser import StackVisualiserView
 from mantidimaging.test_helpers import start_qapplication
 
 versions._use_test_values()
+docks = []
 
 
 @start_qapplication
@@ -38,6 +39,7 @@ class StackVisualiserViewTest(unittest.TestCase):
             self.window = MainWindowView()
         self.window.remove_stack = mock.Mock()
         self.dock, self.view, self.test_data = self._add_stack_visualiser()
+        docks.append(self.dock)
 
     def _add_stack_visualiser(self) -> Tuple[QDockWidget, StackVisualiserView, Images]:
         test_data = th.generate_images()

